lib/link.js

```diff
@@ -97,17 +102,18 @@ export default class Link extends Component {
 
     // If child is an <a> tag and doesn't have a href attribute we specify it so that repetition is not needed by the user
     if (child.type === 'a' && !('href' in child.props)) {
-      props.href = this.props.as || this.props.href
+      props.href = as || (href && typeof href === 'object' ? format(href) : href)
```

Comment:
What happen if `as` is also an object. So, we need to first convert both `as` and `href` into the string version.
That's easy and less error prone.

Do this in the `constructor` and `componentWillReceiveProps` and save it to the `this` context. So, we can use it efficiently from everywhere in the link.